.github/workflows/pr-validation.yml

```diff
@@ -0,0 +1,115 @@
+name: PR Validation
+on:
+  pull_request:
+    types: [ 'opened', 'edited', 'reopened', 'synchronize' ]
+concurrency:
+  group: ${{ github.head_ref }}
+  cancel-in-progress: true
+jobs:
+  pr-validation:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Assign PR to Author
+        uses: samspills/assign-pr-to-author@v1.0
+        with:
+          repo-token: '${{ secrets.PROJECT_GITHUB_TOKEN }}'
+      - name: Lint branch name
+        uses: lekterable/branchlint-action@v1.0.0
+        with:
+          allowed: |
+            main
+            /(feature|bugfix|hotfix)\/\d{2,}.*/i
+            /(release)\/.*/i
+      - name: Title PR Name
+        uses: seferov/pr-lint-action@master
+        with:
+          title-regex: '^#\d{2,}.*'
+          title-regex-flags: 'g' # optional
+          error-message: 'GitHub issue number not found in your title or is invalid'
+      - name: "Build Android project"
+        uses: vgaidarji/android-github-actions-build@v1.0.1
+        with:
+          args: "./gradlew assembleDebug"
+  lint-check:
+    name: Lint checks
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Set up JDK 17
+        uses: actions/setup-java@v1
+        with:
+          java-version: 17
+      - name: Grant execute permission for gradlew
+        run: chmod +x gradlew
+      - name: Run Ktlint
+        run: ./scripts/ktlint.sh -F -DisCI=true
+      - name: Run Detetk
+        run: ./gradlew detekt -DisCI=true
+  android-lint-check:
+    name: Android lint checks
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Set up JDK 17
+        uses: actions/setup-java@v1
+        with:
+          java-version: 17
+      - name: Grant execute permission for gradlew
+        run: chmod +x gradlew
+      - name: Check style
+        run: ./gradlew lintDebug -DisCI=true
+  unit-tests:
+    name: Unit tests
+    runs-on: ubuntu-latest
+    timeout-minutes: 15
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Set up JDK 17
+        uses: actions/setup-java@v1
+        with:
+          java-version: 17
+      - name: Run unit tests
+        run: ./gradlew testDebugUnitTest -DisCI=true
+      - name: Copy test results
+        if: always()
+        run: |
+          mkdir -p junit
+          find . -type f -regex ".*/build/test-results/.*xml" -exec cp {} junit/ \;
+      - name: Upload test results
+        if: always()
+        uses: actions/upload-artifact@v1
+        with:
+          name: junit-results
+          path: junit
+  instrumented-tests:
```

Comment:
Kindly take this out as we may not be doing any UI tests for now. The process for this takes too long to complete.

Reply:
okay